Start and finish project/

Evratal opened the pull request.

Evratal committed:
Start and finich project/

painassasin commented on .idea/Vacancy Management System.iml:
Директорию .idea нужно удалить и добавить в .gitignore
```
git rm --cached .idea/ -r
```

painassasin commented on README.md:
````diff
@@ -0,0 +1,32 @@
+
+   ```bash
+   git clone https://github.com/Evratal/vacancy-management-system.git
+Перейдите в директорию проекта:
````
После команды нужно закрыть кавычки (которые перед bash), причем сделать это нужно на новой строке

painassasin commented on README.md:
```diff
@@ -0,0 +1,32 @@
+cd vacancy-management-system
+Установите необходимые зависимости:
+
+pip install -r requirements.txt
```
Данные команды также нужно оформлять как выше с bash сделал

painassasin commented on src/vacancy.py:
```diff
@@ -0,0 +1,44 @@
+class Vacancy:
+    def __init__(self, id: str, name: str, salary: dict, url: str, employer: dict, snippet: dict):
```
Для класса вакансии нужно использовать `__slots__`

painassasin commented on src/savejson.py:
```diff
@@ -0,0 +1,47 @@
+from src.settings import BASE_DIR
+
+
+class SaveJson:
```
Этот класс в работе не нужен - для работы с вакансиями и json используй `JsonVacancyStorage`

painassasin commented on main.py:
```diff
@@ -0,0 +1,7 @@
+if __name__ == "__main__":
+    hh_api = HH()  # Создаем экземпляр класса HH
+    hh_api.connect()  # Подключаемся к API
+    hh_api.user_interaction()  # Запускаем взаимодействие с пользователем
```
Не вижу реализации HH в реквесте(

Сразу замечу, что user_interaction не должен быть методом HH. У тебя класс HH занимается общением с api, а user_interaction - с пользовтелем через консоль

Нужно сделать этот метод в виде отдельной функции

Evratal committed:
Сделаны поправки.
1. Удалена директория idea и создан gitignore
2. Поправлен Readme файл
3. Добавлены slots в класс Vacancy
4. Удален один из двух модулей для работы с сохранением информации. Функционал перенесен на второй.
5. Метод user_interaction выведен из класса HH и перенесен в отдельный модуль utils.
